contracts/smart-contract-wallet/BaseSmartAccount.sol

```diff
@@ -66,7 +66,10 @@ abstract contract BaseSmartAccount is IAccount, BaseSmartAccountErrors {
             revert CallerIsNotAnEntryPoint(msg.sender);
         validationData = _validateSignature(userOp, userOpHash);
         if (userOp.initCode.length == 0) {
-            _validateAndUpdateNonce(userOp);
+            (bool success, ) = address(entryPoint()).staticcall(
```

Comment:
if it's a success we just skip?

Reply:
alright because entrypoint will validate and update nonce on its own. 

Have we looked into sample wallet implementation changes / making suitable for gnosis as per test cases in the open PR https://github.com/eth-infinitism/account-abstraction/pull/247